microsoft/fhir-server
# Add capability to include supported but not fully indexed search params in search results

## Description
Adds a header which can be included on a search request which allows searching with parameters that are not fully indexed.

## Related issues
Addresses [issue AB#73601].

## Testing
Added Unit tests.  Manually tested with following steps:
1) Run FHIR server to initialize Cosmos DB with __searchparameterstatus__ documents
2) Stop service, and find __searchparameterstatus__ for parameter with uri: "http://hl7.org/fhir/SearchParameter/Patient-identifier"
3) Update parameter status to "Supported" instead of "Enabled"
4) Search for a patient using identifier parameter.  Results should include all patients as identifier parameter is ignored as not enabled
5) Add header "x-ms-use-partial-indices" with value of "true"
6) Repeat search of patient with identifier specified.  Now only matching patient is returned as identifier parameter is honored.
